src/main.ts

```diff
@@ -118,60 +80,39 @@ function resetGameData() {
       },
     },
   });
-  const sizeBuffer = device.createBuffer({
-    size: 2 * Uint32Array.BYTES_PER_ELEMENT,
-    usage:
-      GPUBufferUsage.STORAGE |
-      GPUBufferUsage.UNIFORM |
-      GPUBufferUsage.COPY_DST |
-      GPUBufferUsage.VERTEX,
-    mappedAtCreation: true,
-  });
-  new Uint32Array(sizeBuffer.getMappedRange()).set([
-    GameOptions.width,
-    GameOptions.height,
-  ]);
-  sizeBuffer.unmap();
+  const sizeBuffer = root
+    .createBuffer(vec2u, vec2u(GameOptions.width, GameOptions.height))
+    .$usage("uniform")
+    .$usage("storage");
   const length = GameOptions.width * GameOptions.height;
-  const cells = new Uint32Array(length);
+  const cells = Array.from({ length }).fill(0) as number[];
   for (let i = 0; i < length; i++) {
     cells[i] = Math.random() < 0.25 ? 1 : 0;
   }
```

Comment:
Suggestion, could be simplified a bit more to remove the type cast.
```suggestion
  const cells = Array.from({ length }).map(() => Math.random() < 0.25 ? 1 : 0);
```